test/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/ValueWeightOfGoodsControllerSpec.scala

```diff
@@ -0,0 +1,135 @@
+
+    "redirect to /goods-destination" when {
+      "a declaration has been started but a required answer is missing in the journey" in {
+        givenADeclarationJourneyIsPersisted(startedDeclarationJourney)
+
+        val result = controller.onPageLoad()(request)
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).get mustEqual routes.GoodsDestinationController.onPageLoad().toString
+      }
+    }
+
+    "return OK and render the view" when {
+      "a declaration has been started with a destination of Northern Ireland" in {
+        givenADeclarationJourneyIsPersisted(startedDeclarationJourney.copy(maybeGoodsDestination = Some(NorthernIreland)))
+
+        val result = controller.onPageLoad()(request)
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual
+          view(form, NorthernIreland)(request, messagesApi.preferred(request), appConfig).toString
+      }
+
+      "a declaration has been started with a destination of England, Wales or Scotland" in {
+        givenADeclarationJourneyIsPersisted(startedDeclarationJourney.copy(maybeGoodsDestination = Some(EngScoWal)))
+
+        val result = controller.onPageLoad()(request)
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual
+          view(form, EngScoWal)(request, messagesApi.preferred(request), appConfig).toString
+      }
+
+      "a declaration has been started and a value saved" in {
+        givenADeclarationJourneyIsPersisted(startedDeclarationJourney
+          .copy(
+            maybeGoodsDestination = Some(NorthernIreland),
+            maybeValueWeightOfGoods = Some(true)))
+
+        val result = controller.onPageLoad()(request)
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual
+          view(form.fill(true), NorthernIreland)(request, messagesApi.preferred(request), appConfig).toString
+      }
+    }
+  }
+
+  "onSubmit" must {
+    val url = routes.ValueWeightOfGoodsController.onSubmit().url
+    val postRequest = buildPost(url, sessionId)
+
+    behave like anEndpointRequiringASessionIdAndLinkedDeclarationJourneyToUpdate(controller, url)
+
+    "Redirect to /search-goods" when {
+      "a declaration is started and false is submitted" in {
+        givenADeclarationJourneyIsPersisted(startedDeclarationJourney.copy(maybeGoodsDestination = Some(NorthernIreland)))
+
+        val request = postRequest.withFormUrlEncodedBody(("value", "false"))
+        val result = controller.onSubmit()(request)
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).get mustEqual routes.SkeletonJourneyController.searchGoods().toString
+
+        startedDeclarationJourney.maybeValueWeightOfGoods mustBe None
+        declarationJourneyRepository.findBySessionId(sessionId).futureValue.get.maybeValueWeightOfGoods mustBe Some(false)
+      }
+    }
+
+    "Redirect to /cannot-use-service" when {
```

Comment:
This sounds like something we could abstract into a behaviour similar to anEndpointRequiringASessionIdAndLinkedDeclarationJourneyToUpdate as there are a few redirects to routes.CannotUseServiceController.onPageLoad